ui/frame_tab_view.cpp

```diff
@@ -0,0 +1,136 @@
+
+    m_zoom_in_button = new QPushButton("Zoom In (+)", this);
+    m_zoom_out_button = new QPushButton("Zoom Out (-)", this);
+
+    QHBoxLayout *controls_layout = new QHBoxLayout();
+    controls_layout->addWidget(m_zoom_in_button);
+    controls_layout->addWidget(m_zoom_out_button);
+    controls_layout->addStretch();
+
+    QVBoxLayout *main_layout = new QVBoxLayout(this);
+    main_layout->addLayout(controls_layout);
+    main_layout->addWidget(m_scroll_area);
+
+    QObject::connect(m_zoom_in_button, &QPushButton::clicked, this, &FrameTabView::OnZoomIn);
+    QObject::connect(m_zoom_out_button, &QPushButton::clicked, this, &FrameTabView::OnZoomOut);
+}
+
+// QPixmap does not derive from QObject, so it needs to be manually deleted.
+FrameTabView::~FrameTabView()
+{
+    delete m_image;
+}
+
+void FrameTabView::OnCalculateInitialScale()
+{
+    if (!m_image->isNull() && m_initial_scale_needed)
+    {
+        QSize viewport_size = m_scroll_area->viewport()->size();
+        QSize image_size = m_image->size();
+
+        double width_ratio = static_cast<double>(viewport_size.width()) /
+                             static_cast<double>(image_size.width());
+        double height_ratio = static_cast<double>(viewport_size.height()) /
+                              static_cast<double>(image_size.height());
+
+        // Use the smaller ratio to guarantee the whole image fits.
+        m_scale_factor = qMin(width_ratio, height_ratio);
+
+        ScaleAndDisplayImage();
+
+        // Mark the initial scaling as complete.
+        m_initial_scale_needed = false;
+        m_initial_scale_factor = m_scale_factor;
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+void FrameTabView::OnCaptureScreenshotLoaded(const QString &file_path)
+{
+    if (m_image->load(file_path))
+    {
+        m_scale_factor = 1.0;
+        m_initial_scale_needed = true;
+
+        // Defer the scaling calculation until the event queue has processed
+        QTimer::singleShot(10, this, &FrameTabView::OnCalculateInitialScale);
+    }
+    else
+    {
+        m_image_label->setPixmap(QPixmap());
+        m_image_label->update();
+    }
+}
+
+void FrameTabView::ScaleAndDisplayImage()
+{
+    if (m_image->isNull())
+        return;
+
+    QSize new_size = m_image->size() * m_scale_factor;
+
+    QPixmap scaled_image = m_image->scaled(new_size, Qt::KeepAspectRatio, Qt::SmoothTransformation);
+
+    m_image_label->setPixmap(scaled_image);
+    m_image_label->resize(scaled_image.size());
+}
+
+//--------------------------------------------------------------------------------------------------
+void FrameTabView::OnZoomIn()
+{
+    m_scale_factor *= kZoomStepFactor;
+    ScaleAndDisplayImage();
+}
+
+//--------------------------------------------------------------------------------------------------
+void FrameTabView::OnZoomOut()
+{
+    if (m_scale_factor / kZoomStepFactor > m_initial_scale_factor)
```

Comment:
it would be useful to zoomout a bit more: sometimes the image could be too large to fit, and it would be easier to view when it is smaller than the original size, maybe 0.1 of m_initial_scale_factor?